llvm/include/llvm/Transforms/Utils/LoopUtils.h

```diff
@@ -265,6 +268,21 @@ makeFollowupLoopID(MDNode *OrigLoopID, ArrayRef<StringRef> FollowupAttrs,
                    const char *InheritOptionsAttrsPrefix = "",
                    bool AlwaysNew = false);
 
+/// Update the minimum and maximum iteration count of a loop
+///
+/// @param Context the context to create new MDNodes in
+///
+/// @param LoopID the loop id to update
+///
+/// @param FixMin operation to apply to min_iteration_count
+///
+/// @param FixMax operation to apply to max_iteration_count
+///
+/// @return The updated LoopID
+MDNode *updateIterCounts(LLVMContext &Context, MDNode *LoopID,
+                         std::function<int64_t(int64_t)> FixMin,
+                         std::function<int64_t(int64_t)> FixMax);
```

Comment:
Hmm. Min/MaxIterCount is our extension, and we have updated a few loopengines to repair it. I expect discussions about the detailed semantics ad whether it can be invalidated by non-loop-ware engines.